app/lib/travis/TravisApiClient.scala

```diff
@@ -19,114 +18,147 @@ class TravisApiClient(githubToken: String) extends LazyLogging {
+      readResponseFrom(new Builder().url(baseEndpoint + path)
+        .headers(headersContainer(DefaultJsonApiHeaders))
+        .get()
+        .build())
+    }
 
-  case class AuthResponse(access_token: String)
+    def postJson[Req, Resp](path: String, reqBody: Req, extraHeaders: Seq[(String, String)] = Seq.empty)(implicit wReq: Writes[Req], rResp: Reads[Resp]): Future[JsResult[Resp]] = {
+      readResponseFrom(new Builder().url(baseEndpoint + path)
+        .headers(headersContainer(DefaultJsonApiHeaders ++ extraHeaders))
+        .post(RequestBody.create(JsonMediaType, toJson(reqBody).toString))
+        .build())
+    }
 
-  case class Build(id: Int, config: JsValue, state: String)
+    def headersContainer(headers: Seq[(String, String)]) = Headers.of(headers.toMap.asJava)
 
-  object Build {
-    implicit val readsBuild = Json.reads[Build]
-  }
+    def readResponseFrom[Resp](request: Request)(implicit rResp: Reads[Resp]): Future[JsResult[Resp]] = {
+      logger.info(s"${request.method()} - ${request.url.encodedPath}")
+      okHttpClient.execute(request) { resp => Json.parse(resp.body().byteStream()).validate[Resp] }
+    }
 
-  case class Commit(sha: String, message: String)
+    case class AuthRequest(github_token: String)
 
-  object Commit {
-    implicit val readsCommit = Json.reads[Commit]
-  }
+    object AuthRequest {
+      implicit val writesAuthRequest = Json.writes[AuthRequest]
+    }
 
-  case class BuildResponse(build: Build, commit: Commit)
+    case class AuthResponse(access_token: String)
 
-  object BuildResponse {
-    implicit val readsBuildResponse = Json.reads[BuildResponse]
-  }
+    case class Build(id: Int, config: JsValue, state: String)
 
-  object AuthResponse {
-    implicit val readsAuthRequest = Json.reads[AuthResponse]
-  }
+    object Build {
+      implicit val readsBuild = Json.reads[Build]
+    }
 
-  def auth(githubToken: String) = postJson[AuthRequest, AuthResponse]("/auth/github", AuthRequest(githubToken))
+    case class Commit(sha: String, message: String)
 
-  def build(buildId: String) = getJson[BuildResponse](s"/builds/$buildId")
+    object Commit {
+      implicit val readsCommit = Json.reads[Commit]
+    }
 
-  def requestBuild(repoId: String, travis: TravisCI, message: String, buildBranch: String) = {
-    val bodyJson = Json.obj(
-      "request" -> Json.obj(
-        "message" -> message,
-        "branch" -> buildBranch,
-        "config" -> travis.config
-      )
-    )
+    case class BuildResponse(build: Build, commit: Commit)
+
+    object BuildResponse {
+      implicit val readsBuildResponse = Json.reads[BuildResponse]
+    }
+
+    object AuthResponse {
+      implicit val readsAuthRequest = Json.reads[AuthResponse]
+    }
+
+    def auth(githubToken: String) = postJson[AuthRequest, AuthResponse]("/auth/github", AuthRequest(githubToken))
 
-    for {
+    def build(buildId: String) = for {
       authToken <- authTokenSupplier.get()
```

Comment:
This request doesn't _require_ auth headers for the Travis CI API, but it does for the Travis Pro API. As we're using token refresh anyway, we just pass the auth header regardless of the environment.